rbac/internal/views.py

```diff
@@ -338,6 +341,155 @@ def get_org_admin(request, org_or_account):
     return HttpResponse('Invalid method, only "GET" is allowed.', status=405)
 
 
+def user_lookup(request):
+    """Get all groups, roles, and permissions for a provided user via username or email.
+
+    If both params are provided, email is ignored and username is used.
+
+    GET /_private/api/utils/user_lookup/?username=foo&email=bar@redhat.com
+    """
+    if request.method != "GET":
+        return handle_error("Invalid http method - only 'GET' is allowed", 405)
+
+    username = request.GET.get("username")
+    email = request.GET.get("email")
+
+    try:
+        validate_user_lookup_input(username, email)
+    except ValueError as err:
+        return handle_error(f"Invalid request input - {err}", 400)
+
+    try:
+        user = get_user_from_bop(username, email)
+    except UserNotFoundError as err:
+        return handle_error(f"Not found - {err}", 404)
+    except Exception as err:
+        return handle_error(f"Internal error - couldn't get user from bop: {err}", 500)
+
+    result = {
+        "username": user["username"],
+        "email_address": user["email"],
+    }
+
+    try:
+        principals = Principal.objects.filter(username__iexact=user["username"])
```

Comment:
We don't have username indexed currently, so this query might be slow. Better to get the org_id from the response and get the tenant and combine the query (we have unique index for username and tenant)

Reply:
awesome, yeah I'll absolutely do that. Ty for the help :heart: I didn't realize I could just snag the org id from the bop user since we are querying for that already. Makes total sense

Reply:
ok reworked this to query for username and tenant now. Get the tenant here: https://github.com/RedHatInsights/insights-rbac/pull/1569/files#diff-fb45a5441f65035cd9b020f4b00090da2927a3b9c40c9dc56848e7129c9219ceR382

and then call this `utils.get_principal` method here: https://github.com/RedHatInsights/insights-rbac/pull/1569/files#diff-fb45a5441f65035cd9b020f4b00090da2927a3b9c40c9dc56848e7129c9219ceR390

which is doing the same query I was, but with the tenant:
https://github.com/RedHatInsights/insights-rbac/blob/master/rbac/management/utils.py#L103

and this method has the added bonus of creating the principal if not found in RBAC despite the user existing in BOP